src/components/CommonComponents/Pagination/index.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { FC, MouseEvent, useCallback, useMemo } from 'react';
+import { buildPaginationModel } from './util/buildPaginationModel';
+import Page from './Page';
+import styles from './index.module.scss';
+
+interface Props {
+  currentPage: number;
+  hrefBuilder?: (n: number) => string;
+  marginPageCount: number;
+  pageCount: number;
+  onPageChange?: (e: MouseEvent, n: number) => void;
+  showPages?: boolean;
+  surroundingPageCount?: number;
+  wrapperClassName?: string;
+}
+
+const defaultHrefBuilder = (pageNum: number): string => `#${pageNum}`;
+
+const Pagination: FC<Props> = ({
+  pageCount,
+  currentPage,
+  onPageChange,
+  hrefBuilder = defaultHrefBuilder,
+  marginPageCount = 1,
+  showPages = true,
+  surroundingPageCount = 2,
+  wrapperClassName,
+}) => {
+  const pageChange = useCallback(
+    (n: number) => (e: MouseEvent) => onPageChange && onPageChange(e, n),
+    [onPageChange]
+  );
+
+  const model = useMemo(
```

Comment:
Not sure memoization here will have any benefit with the amount of arguments we have here. You could just make one single memoized result that already includes your callback, pagination model and the mapped JSX Elements. That would be "simpler".

Reply:
fixed